x-pack/plugins/apm/public/components/app/tutorials/commands/dotnet.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+
+export const dotnetVariables = (apiKey?: string) => ({
+  apmServiceName: 'ServiceName',
+  ...(!apiKey && { secretToken: 'SecretToken' }),
+  ...(apiKey && { apiKey: 'ApiKey' }),
+  apmServerUrl: 'ServerUrl',
+  apmEnvironment: 'Environment',
+});
+
+export const dotnetHighlightLang = 'dotnet';
+
+export const dotnetLineNumbers = (apiKey?: string) => ({
+  start: 1,
+  highlight: '1-2, 4, 6, 8, 10-12',
+});
+
+export const dotnet = `{
+  "ElasticApm": {
+    /// {{serviceNameHint}} ${i18n.translate(
+      'xpack.apm.tutorial.dotnetClient.createConfig.commands.defaultServiceName',
+      {
+        defaultMessage: 'Default is the entry assembly of the application.',
+      }
+    )}
+    "ServiceName": "my-service-name",
+    {{#apiKey}}
+    /// {{apiKeyHint}}
+    "ApiKey": "{{{apiKey}}}",
+    {{/apiKey}}
+    {{^apiKey}}
+    /// {{secretTokenHint}}
+    "SecretToken": "{{{secretToken}}}",
+    {{/apiKey}}
+    /// {{{serverUrlHint}}}
+    "ServerUrl": "{{{apmServerUrl}}}",
+    /// {{{serviceEnvironmentHint}}}
+    "Environment": "my-environment",
```

Comment:
Can you bring up this format with the agent developers? I think you should also show them an alternative format where instead of having string variables, the strings are duplicated and not translated. 
We should make sure that what we are building also works for the agent devs since they will have to maintain the docs.

Reply:
I don't understand what you mean. The idea is Agent devs should be easily be able to contribute to the doc changes here.

There are 2 possibilities
1. All the code comments for commonly used variables are stored in a single place here - `x-pack/plugins/apm/public/components/app/tutorials/commands/shared_hints.ts`. Which makes it easier for them to edit
2. Agent specific changes can be directly hardcoded in these files as its done for some agents.

Reply:
@chrisdistasio Quick question here, Do you think the code comments which we produce here must be translated ?

Reply:
"translated" as in localization for (spoken) language?

Reply:
Had a discussion with Chris and he pointed out that since we were supporting translations upto now, we must not change it. Also from Agent Developers perspective, updating these snippets should be fairly easy